### ISSUE TITLE
Calling <Plug>(coc-references) causes list to appear over, and entire menu is bugged

### ISSUE THREAD

**Issue by unphased** [neoclide/coc.nvim#3005]
I was dealing with this for months, and I suspect it's something wrong with my configuration. 

Today I found good evidence for what is being run that is bad. In my CocInfo logs: 

```
2021-04-02T00:11:31.933 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:32.686 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:33.103 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:33.771 INFO (pid:30638) [attach] - receive notification: jumpReferences []
2021-04-02T00:11:33.776 INFO (pid:30638) [attach] - receive notification: openList [ '--normal', '--auto-preview', 'location' ]
2021-04-02T00:11:45.916 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:47.044 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:47.477 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:48.467 INFO (pid:30638) [attach] - receive notification: jumpDefinition []
2021-04-02T00:11:48.669 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:51.563 INFO (pid:30638) [attach] - receive notification: jumpDefinition []
2021-04-02T00:11:51.766 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:55.472 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:11:56.768 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:12:03.591 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:12:06.186 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:12:12.723 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:12:13.206 INFO (pid:30638) [attach] - receive notification: highlight []
2021-04-02T00:12:13.484 INFO (pid:30638) [attach] - receive notification: jumpReferences []
2021-04-02T00:12:13.490 INFO (pid:30638) [attach] - receive notification: openList [ '--normal', '--auto-preview', 'location' ]
```

See how every call to jumpReferences is followed immediately by openList. It's this openlist (and those flags) that I can see in the menu which comes up right after the References list flashes for one frame. It is broken, the cursor is broken, pressing arrow keys and stuff like that will often cause me to drop back into the buffer with a broken missing cursor, it's really bad. 

I just want to figure out why it's being triggered by jumpReferences. Sometimes, usually, my jumpreference call works fine. I still havent been able to work out exactly what situations cause this second bad list to appear. 

This only seems to happen with references requested via clangd. I also use tsserver, never seen that from there.

The map is just `nmap <silent> gr <Plug>(coc-references)` exactly like shown in the readme.

**Comment by chemzqm** [comment]
`:h CocLocationsChange`

**Comment by unphased** [comment]
Thanks, I will check that. 

Meanwhile: Here is the asciinema. You can see the problem when the cursor is flashing between the bottom command line and the menu, when the weird erroneous menu is up. 

https://asciinema.org/a/rvDvwkeTLbm8fczsWWToe7Psc

**Comment by chemzqm** [comment]
`:h g:coc_disable_transparent_cursor` the hidden cursor logic would be buggy with your neovim on some terminal.

**Comment by unphased** [comment]
I'm not using neovim because it's not stable. But I built this vim from source fairly recently. This has been an issue for a very long time. I update all my plugins regularly in hopes this will go away. 

When I set `let g:coc_enable_locationlist = 0` in my vimrc, now `coc-references` never opens any menu :(

**Comment by unphased** [comment]
When I set 

```
let g:coc_enable_locationlist = 0
autocmd User CocLocationsChange CocList --normal location
```
This seems to be working much better now. Do you know why the default setting may break like I was seeing?

**Comment by unphased** [comment]
> `:h g:coc_disable_transparent_cursor` the hidden cursor logic could be buggy with your neovim on some terminal.

Sure this would only be addressing the symptom. When the list menu breaks as i describe, as you see in my asciinema recording, it actually dumps terminal codes in as normal mode and drops me in insert mode with some garbage. I clearly need to fix the root cause of the weird double menu spawning.

**Comment by unphased** [comment]
I'm still able to get weird double menu spawning. It's slightly less broken than before. The visual aspect of it is ok (doesnt look totally wrong) but one way to tell it's in a broken state is by using arrow keys or hitting `p` to toggle preview, which breaks the menu.

I will likely start to instrument coc.nvim plugin code, it should be possible to track down what's going on at some point.

**Comment by unphased** [comment]
Great, before i got into instrumenting I decided to bisect by deleting sections of my vimrc. Upon erasing all my other plugins and my vimrc this does not reproduce. So that's a lead and I feel a lot better now. Will definitely report back with the root cause. Might take a few days to get through it though.

**Comment by unphased** [comment]
I have come to diagnose the problem, as it has continued happening. I found via manual binary culling in my vimrc that the issue no longer reproduces once I removed the setting `set noequalalways` from it. Something about the custom menu that coc builds is broken by `set noequalalways`. I hope that this helps and may jog your spidey-senses as to why it might cause the strange stuff I was getting. For now, I will just make a clear comment in my config and abandon the usage of this setting.
